### ISSUE TITLE
Tables

### ISSUE THREAD

**Issue by jackcmeyer** [HospitalRun/components#28]
## 🚀 Feature Proposal

Tables are used to display lists of information in an organized way.

A table should be able to: 

- [ ] pagination
- [ ] fuzzy search
- [ ] searchable columns
- [ ] sortablecontent
- [ ] custom column actions
- [ ] injectable behaviors
- [ ] inline row details

## Motivation

Tables are core to information displaying. 

## Example

Listing patients, appointments, diagnoses, 

## Relevant Links
https://www.npmjs.com/package/react-table

**Comment by fox1t** [comment]
@irvelervel I think we can start discussing about tables. Let's find the best component lib to start from. Some of the features we need are server side pagination, filtering, sorting and probabily inline editing. We need to support advanced filters, so we can express all of the possibilities of https://github.com/fox1t/qs-to-mongo

**Comment by ocBruno** [comment]
I was going to start the boilerplate for this but wasn't sure if the discussion points had been decided upon so I tried doing a little research. I found two libraries that might be good solutions.

I thought we could use this react data grid library and handle the server side pagination ourselves,
https://adazzle.github.io/react-data-grid

.. or this react bootstrap table library which already has a remote option for server side fetching
https://react-bootstrap-table.github.io/react-bootstrap-table2/storybook/

I'm not sure they fit the needs of the table component but give me a heads up if so and I'll give this issue a go!

**Comment by jackcmeyer** [comment]
@ocBruno Any reason react-tables isn't in this list? I think this is what the team  had in mind for the table component. All thought, if you think there is a better solution, we'd love to hear it!

**Comment by ocBruno** [comment]
> @ocBruno Any reason react-tables isn't in this list? I think this is what the team had in mind for the table component. All thought, if you think there is a better solution, we'd love to hear it!

@jackcmeyer Thanks for the feedback! I'm not sure how I missed this one lol, it does seem to be the best solution! 

I'm going to begin attempting to bootstrap this component with react tables and some of the basic necessary features. 

I am still not too experienced with react and typescript but have been studying and I will post my progress in case anybody wants to give me some feedback or spot me on a mistake!

**Comment by irvelervel** [comment]
Hello everyone. I made a wrapper some months ago for an older version of react-table (v6), which was more like a standard table library (pretty vast in functionality, especially in manual mode, and provided with a built-in UI), but for the new HospitalRun we was internally more oriented to go with the last version (v7) and I decided to scrap the older work. There are many reasons that pushed us for this decision: primarily the fact that react-table v7 it's headless, which means we can adapt it to every UI we are going to build (and maybe update in the future). It also now uses hooks, which may be intimidating but provide much faster performances. Thanks @ocBruno for the research and the ideas :smile: I had to delay this task but now I should have more time to work on it.
